cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -456,9 +463,15 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         }
 
         Future<ReconciliationState> zkRollingUpdate() {
-            return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(), pod ->
-                isPodToRestart(zkDiffs.resource(), pod, zkAncillaryCmChange, this.clusterCa)
-            ));
+            return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(), pod -> {
+                boolean isPodToRestart = isPodToRestart(zkDiffs.resource(), pod, zkAncillaryCmChange, this.clusterCa);
+                boolean isSatisfiedBy = true;
+                // it makes sense to check maintenance windows if pod restarting is needed
+                if (isPodToRestart) {
+                    isSatisfiedBy = isMaintenanceTimeWindowsSatisfied();
```

Comment:
Well, let me try to explain ...
The way we roll ZK or K and the EO are different as you know. We have our logic for ZK and K but we are relying on native Kuvernetes rolling update for EO (applying some patch on the resource).
It's true that `isPodToRestart` evaluates if a ZK or K pod needs a restart for some reasons and only, in that case, it makes sense to check if the maintenance time windows are satisfied otherwise it's a useless call because in any case, we are not going to roll the pod.
In case of EO we are just calling the reconcile leaving the native Kubernetes support to execute rolling if something changes. Before that we don't know if the desired deployment is different from current one (see this as a potential `isPodToRestart`) so the reason why I call in any case the check on the maintenance time windows (which is useful to get the right cert generation to put on the deployment). 
What you say means that we should get a way for knowing early that desired deployment is different or not from the current one but consider that the cert generation annotation is already always a difference, this is the reason why this method put the annotation in any case (but now I needed to put the right one, current or new cert genaration).